internal/command/test.go

```diff
@@ -235,14 +235,20 @@ func (c *TestCommand) Run(rawArgs []string) int {
 			TestingDirectory:    args.TestDirectory,
 			Opts:                opts,
 			View:                view,
-			JUnit:               junitFile,
 			Stopped:             false,
 			Cancelled:           false,
 			StoppedCtx:          stopCtx,
 			CancelledCtx:        cancelCtx,
 			Filter:              args.Filter,
 			Verbose:             args.Verbose,
 		}
+
+		if junitFile != nil {
+			junitFile.SetTestSuiteRunner(localRunner)
+			localRunner.JUnit = junitFile
```

Comment:
```suggestion
			if args.JUnitXMLFile != "" {
			    localRunner.JUnit = junit.NewTestJUnitXMLFile(args.JUnitXMLFile, c.configLoader, localRunner)
```

Something like this?